Kernel/Devices/Audio/AC97/AC97.cpp

```diff
@@ -35,6 +35,15 @@ UNMAP_AFTER_INIT ErrorOr<NonnullRefPtr<AudioController>> AC97::create(PCI::Devic
 UNMAP_AFTER_INIT ErrorOr<bool> AC97::probe(PCI::DeviceIdentifier const& device_identifier)
 {
     VERIFY(device_identifier.class_code().value() == to_underlying(PCI::ClassID::Multimedia));
+
+    // If BAR0 is 64-bit then BAR1 isn't usable
+    u64 pci_bar0_value = PCI::get_BAR(device_identifier, PCI::HeaderType0BaseRegister::BAR0);
+    if (PCI::get_BAR_space_type(pci_bar0_value) == PCI::BARSpaceType::Memory64BitSpace)
+        return Error::from_errno(EIO);
+
+    if (PCI::get_BAR_space_size(device_identifier, PCI::HeaderType0BaseRegister::BAR1) < 0x30)
```

Comment:
What is this constant `0x30` representing? Can you give it a proper description?

Reply:
This driver needs registers up to [2c...2f]. So anything with a shorter bar1 is definitely incompatible. Renamed the const.

Reply:
Ah so I see what you are trying to do - checking the BAR sizes seems sensible for a probe, but I have two suggestions:

* Check both BAR0 and BAR1 sizes. We know which registers should be in both.
* Instead of adding a random new constexpr, maybe you can add a new entry to the ends of both `AC97::NativeAudioBusRegister` and `AC97::NativeAudioMixerRegister` so it's immediately clear you're comparing the BAR sizes to the registers they contain.

Reply:
Done